analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs

```diff
@@ -77,9 +77,9 @@ public void Nullable_HasValue_ReadsBoolConstraintFromObjectConstraint()
             Tag("SymbolAfterNull", arg);
             """;
         var validator = SETestContext.CreateCS(code, ", bool? arg").Validator;
-        validator.ValidateTag("HasValueUnknown", x => x.Should().BeNull());
+        validator.ValidateTag("HasValueUnknown", x => x.HasConstraint<BoolConstraint>().Should().BeFalse());
         validator.ValidateTag("SymbolUnknown", x => x.Should().BeNull());
-        validator.ValidateTag("HasValueAfterTrue", x => x.Should().BeNull());   // ToDo: Should be x.HasConstraint(BoolConstraint.True).Should().BeTrue()); after we build NotNull for bool literal.
+        validator.ValidateTag("HasValueAfterTrue", x => x.HasConstraint<BoolConstraint>().Should().BeFalse());   // ToDo: Should be x.HasConstraint(BoolConstraint.True).Should().BeTrue()); after we build NotNull for bool literal.
```

Comment:
This is suspicious. I was expecting the `ToDo` go away and have different assertion here

Reply:
Added as task to review later